cmd/storage-rest-server.go

```diff
@@ -133,28 +133,29 @@ func (s *storageRESTServer) DeleteVolHandler(w http.ResponseWriter, r *http.Requ
 	}
 }
 
-// PrepareFileHandler - fallocate() space for a file.
-func (s *storageRESTServer) PrepareFileHandler(w http.ResponseWriter, r *http.Request) {
+// AppendFileHandler - append to a file.
+func (s *storageRESTServer) AppendFileHandler(w http.ResponseWriter, r *http.Request) {
 	if !s.IsValid(w, r) {
 		return
 	}
 	vars := mux.Vars(r)
 	volume := vars[storageRESTVolume]
 	filePath := vars[storageRESTFilePath]
-	fileSizeStr := vars[storageRESTLength]
-	fileSize, err := strconv.Atoi(fileSizeStr)
+
+	buf := make([]byte, r.ContentLength)
+	_, err := io.ReadFull(r.Body, buf)
 	if err != nil {
 		s.writeErrorResponse(w, err)
 		return
 	}
-	err = s.storage.PrepareFile(volume, filePath, int64(fileSize))
+	err = s.storage.AppendFile(volume, filePath, buf)
 	if err != nil {
 		s.writeErrorResponse(w, err)
 	}
 }
 
-// AppendFileHandler - append to a file.
-func (s *storageRESTServer) AppendFileHandler(w http.ResponseWriter, r *http.Request) {
+// CreateFileHandler - fallocate() space for a file.
```

Comment:
Not only allocates also copies the content so comment should reflect that

Reply:
copy/paste problem from comment of PrepareFile, will fix it